pkg/provider/azure_controller_common.go

```diff
@@ -256,6 +262,11 @@ func (c *controllerCommon) AttachDisk(ctx context.Context, async bool, diskName,
 		async:   async,
 	}
 
+	err := c.cloud.SetDiskLun(nodeName, diskToAttach)
+	if err != nil {
+		return -1, err
+	}
+
```

Comment:
You should defer a removal of the temp disk allocation here, but only remove if `err != nil`. The calls on lines 270 and 277 call fail. If they do, the batch function won't be called to remove the temp allocation.